lib/diagnostic.js

```diff
@@ -5,17 +5,23 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
+const min = function (previous, current) {
+  let previous = array[0];
+    array slice(1).forEach(function (e, i, a){
+      current(previous, e, i + 1, array);
+    });
+    return previousl
+  };
+
 
-};
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let total = array.reduce(sum);`
-const sum = function (/* previous, current */) {
-
+const sum = function (previous, current ) {
+let previous = array[0] = array.reduce(sum)
 };
```

Comment:
Actually, this question is _much_ simpler than what you thought it was. Remember, a `callback` is a function is that is called _by_ another function.
```
const sum = function (previous, current) {
  return previous + current;
```